e2e/test-runner/test_utils.go

```diff
@@ -49,36 +44,52 @@ func wslSetup(t *testing.T) {
 		}
 	})
 
+}
+
+// subiquityLogs prints the logs if a test fails
+func subiquityLogs(t *testing.T) string {
+	rootDir := os.Getenv(constants.LauncherRepoEnvVar)
+	clientLogFullPath := filepath.Join(rootDir, clientLogPath)
+
 	// Prints debug logs
-	t.Cleanup(func() {
-		if !t.Failed() {
-			return
-		}
-		t.Log("\n\n==== Subiquity answer file  ====")
-		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-		defer cancel()
-		out, err := wslCommand(ctx, "cat", subiquityAnswerFile).CombinedOutput()
-		t.Logf("%s", out)
-		if err != nil {
-			t.Logf("Failed to retrieve subiquity answer file: %v", err)
-		}
+	var s string
 
-		t.Log("\n\n==== Server Debug Log ====")
-		ctx, cancel = context.WithTimeout(context.Background(), 30*time.Second)
-		defer cancel()
-		out, err = wslCommand(ctx, "cat", ServerLogPath).CombinedOutput()
-		t.Logf("%s", out)
-		if err != nil {
-			t.Logf("Failed to retrieve server debug log: %v", err)
-		}
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	out, err := wslCommand(ctx, "cat", subiquityAnswerFile).CombinedOutput()
+	if err != nil {
+		t.Logf("Failed to retrieve subiquity answer file: %v\n%s", err, out)
```

Comment:
Don’t print out if there is an error. The first argument content validity is not ensure in generally in Go when a function returns an error

Reply:
I know it's bad in most other contexts, but isn't this quite standard for `Cmd.CombinedOutput`? What's the idiomatic way of reading the error message of a failed command?

Reply:
Ah sorry, this is `CombinedOutput()`, so yeah, this is one exception when we don’t redirect stderr before. So, let’s keep it this way.